vowpalwabbit/global_data.h

```diff
@@ -534,8 +535,12 @@ struct vw
   size_t passes_complete;
   uint64_t parse_mask;  // 1 << num_bits -1
   bool permutations;    // if true - permutations of features generated instead of simple combinations. false by default
+
+  // Referenced by examples as their set of interactions. Can be overriden by reductions.
   std::vector<std::string> interactions;
+  // TODO #1863 deprecate in favor of only interactions field.
```

Comment:
Can this be done in a computationally efficient manner?

Reply:
Which bit? I don't follow

Reply:
Shifting quadratics and cubics into interactions.  That was nontrivial the first time, which is why it did not happen.

Reply:
Ah I see, I am not sure of the efficiencies of being able to do this. I more mean to maintain only a single vector rather than 3. When generating the interactions, we can treat the cases of size 2,3 and n differently to maintain the current efficiencies.

Having pairs, triples and interactions kind of leaks this information and I assume is there from before we could handle arbitrary length?

Reply:
They are, yes.